src/services/users.services.js

```diff
@@ -0,0 +1,50 @@
+// const { ApiError } = require('../exceptions/ApiError');
+const { User } = require('../models/user');
+const { hashPassword } = require('../utils/password/hashPassword');
+
+const getUsers = async () => {
+  const users = await User.findAll();
+
+  return users;
+};
+
+const getUserByEmail = async (email) => {
+  const user = await User.findOne({ where: { email } });
+
+  return user;
+};
+
+const getUserById = async (id) => {
+  const user = await User.findOne({ where: { id } });
+
+  return user;
+};
+
+const removeUser = async (id) => {
+  await User.destroy({ where: { id } });
+};
+
+const updateUser = async (user, userData) => {
+  for (const [field, data] of Object.entries(userData)) {
+    if (data && field === 'password') {
+      user[field] = await hashPassword(data);
+    } else if (data) {
+      user[field] = data;
+    }
+  }
+  await user.save();
+
+  const updatedUser = await getUserById(user.id);
+
+  return updatedUser;
```

Comment:
Ensure that the `updateUser` function handles cases where `userData` might not contain any fields to update. Consider adding a check to avoid unnecessary database operations if `userData` is empty or contains no changes.